shared/fa2_modules/test/simple_admin_wrapper.mligo

```diff
@@ -0,0 +1,34 @@
+(*
+  Wrapper around simple_admin for testing:
+  embeds simple_admin.mligo and includes `unit` entrypoints
+  that trigger `fail_if_not_admin`, `fail_if_paused`
+*)
+
+#include "../simple_admin.mligo"
+
+type wrapper_storage = {
+  admin : simple_admin_storage;
+}
+
+type wrapper_param =
+  | Admin of simple_admin
+  | Fail_if_not_admin of unit
+  | Fail_if_paused of unit
+
+let wrapper_main
+    (param, s : wrapper_param * wrapper_storage)
+    : (operation list) * wrapper_storage =
+  match param with
+  | Admin p ->
+      let ops, admin = simple_admin (p, s.admin) in
+      let new_s = { s with admin = admin; } in
+      (ops, new_s)
+
+  | Fail_if_not_admin p ->
+      let u = fail_if_not_admin s.admin in
+      (([]: operation list), s)
+
+  | Fail_if_paused p ->
+      let u = fail_if_paused s.admin in
```

Comment:
I suggest restricting the type of the return value i.e. `let u : unit = fail_if_paused s.admin` given that the return value is never used. If `fail_if_paused` as well as `fail_if_admin` are made functions of two arguments then `fail_if_paused s.admin` will be treated as a partial application and the desired check will not occur.

Reply:
Otherwise LGTM